### PR TITLE
Tested Nvidia Containerfile 

### PR DESCRIPTION
**Pull request by bmahabirbu**
There is still one non-important issue I'm working on. The issue revolves around using cmake when compiling llama.cpp with cuda. The build cannot find the libcuda library 

This could be due to my dev environment. I'm using WSL2 with Nvidia cuda toolkit and container toolkit. 
https://github.com/abetlen/llama-cpp-python/issues/1169.

When switching to Make everything works as intended

### CODE REVIEW

File: container-images/ramalama/latest-cuda/Containerfile [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    make -j $(nproc) GGML_CUDA=1 CUDA_ARCH=ALL && \
+    mv llama-cli /usr/bin/llama-cli && \
+    mv llama-server /usr/bin/llama-server && \
+    cd / && \
+    rm -rf llama.cpp
+
+RUN git clone https://github.com/ggerganov/whisper.cpp.git && \
+    cd whisper.cpp && \
+    git reset --hard ${WHISPER_CPP_SHA} && \
+    make -j $(nproc) && \
```
Comment (ericcurtin) [inline container-images/ramalama/latest-cuda/Containerfile:50]: Can we build whisper.cpp with GGML_CUDA=1 CUDA_ARCH=ALL also?

Comment (bmahabirbu) [reply container-images/ramalama/latest-cuda/Containerfile:50]: Sure!

**Review by ericcurtin — APPROVED** [review]

**Comment by ericcurtin** [comment]
Die to legal restrictions I'm not sure we can push Nvidia/cuda images to quay.io . Nonetheless this will be very useful for those who want to build themselves and try it out!

**Comment by ericcurtin** [comment]
@bmahabirbu do we need -ngl 99 when running with an Nvidia card for accelerated workloads? I don't own an Nvidia GPU...

**Comment by ericcurtin** [comment]
We must get container_build.sh to skip building this CUDA image in CI. This is what I get for merging something just before bedtime!

**Comment by ericcurtin** [comment]
Oh it's the commit was never signed... Please sign your commit next time @bmahabirbu 

git commit -s

is open way of doing so.

**Comment by bmahabirbu** [comment]
> @bmahabirbu do we need -ngl 99 when running with an Nvidia card for accelerated workloads? I don't own an Nvidia GPU...

Unfortunately, I don't believe one number will always work. I had to figure out the number for ngl by incrementing it till the VRAM was almost filled up. I'll keep investigating this further

**Comment by ericcurtin** [comment]
Yeah... 99 means max it out essentially... I think we ought to do that... Maybe introduce a -ngl flag to ramalama if people really want to tweak it manually...

**Comment by bmahabirbu** [comment]
> Oh it's the commit was never signed... Please sign your commit next time @bmahabirbu
> 
> git commit -s
> 
> is open way of doing so.

I'm sorry! Noted

**Comment by ericcurtin** [comment]
@bmahabirbu would appreciate if you could write something like:

docs/cuda.md

docs/wsl2.md

to describe how you got WSL2 with Nvidia cuda toolkit and container toolkit working.

**Comment by bmahabirbu** [comment]
> @bmahabirbu would appreciate if you could write something like:
> 
> docs/cuda.md
> 
> docs/wsl2.md
> 
> to describe how you got WSL2 with Nvidia cuda toolkit and container toolkit working.

Sure, I'm on it

**Pull request merged**